README.md

```diff
@@ -1,35 +1,173 @@
 # Summa Aggregation
 
+Summa Aggregation is focused on optimizing the generation of Merkle sum trees, a task identified as the primary time-consuming process in Summa benchmarks. Our goal is to significantly reduce the time required to generate these trees by leveraging parallelization across multiple machines.
+
+The system features `AggregationMerkleSumTree`, a specialized component designed for the efficient construction of complete trees from smaller, aggregated structures known as `mini-trees`. This approach not only speeds up the tree-building process but also enhances scalability and performance in large-scale data environments.
+
 ## Orchestrator
 
-WIP
+The Orchestrator component will manage the orchestration of multiple `Executors`. Its primary role will be to dynamically allocate tasks and manage the data workflow among the `Executors`.
+
+## Executor and ExecutorSpawner
+
+The `Executor` plays a crucial role in the data processing pipeline. It is responsible for receiving parsed data entries and processing them through the `mini-tree-generator`.
+Each `Executor` operates alongside a `mini-tree-generator`, accessed via a URL spawned in a container by the `ExecutorSpawner`.
```

Comment:
I don't find any mention of the `mini-tree-generator` in the code. It would be clear if you refer to that using the actual name of the function/data structure

Reply:
I moved into `src/mini-tree-server.rs` from `bin/mini-tree-server.rs`